src/test/java/pages/CartPage.java

```diff
@@ -36,7 +40,40 @@ public List<Item> getItemsInCart() {
             String desc = element.findElement(INVENTORY_ITEM_DESCRIPTION).getText();
             Item item = new Item(name, desc, price);
             items.add(item);
+
+        }
+        return items;
+    }
+
+    public int getCartCount() {
+        return Integer.parseInt(driver.findElement(CART_ITEM_COUNT).getText());
+    }
+
+    public List<Item> removeItemFromList() {
+        List<WebElement> elements = driver.findElements(CART_ITEM);
+        List<Item> items = new ArrayList<>();
+        for (WebElement element : elements) {
+            if (element.equals(elements.get(0))) {
+                element.findElement(REMOVE_ITEM_BUTTON).click();
+            } else {
+                String name = element.findElement(INVENTORY_ITEM_NAME).getText();
+                double price = Double.parseDouble(element.findElement(INVENTORY_ITEM_PRICE).getText().replace("$", ""));
+                String desc = element.findElement(INVENTORY_ITEM_DESCRIPTION).getText();
+                Item item = new Item(name, desc, price);
+                items.add(item);
+            }
         }
         return items;
     }
+
+    public void deleteCartItem(int itemNumber) {
```

Comment:
This works ok if you switch it like this:
List<WebElement> cartItems = driver.findElements(REMOVE_ITEM_BUTTON);
cartItems.get(itemNumber).click();

Still not sure why, but will take some time to debug further.